flask-homework/templates/base.html

```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <link rel="stylesheet" href="{{ url_for("static", filename="css/style.css") }}">
+<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-9ndCyUaIbzAi2FUVXJi0CjmCapSmO7SnpJef0486qhLnuZ2cdeRhO02iuK6FUUVM" crossorigin="anonymous">
+    <title>
+        {% block title %}
+            Users and books
+        {% endblock %}
+    </title>
+</head>
+<body>
+    <div class="container">
+        <div class="header">
+            {% block user %}
+                {% if "username" in session %}
+                    <div>Hello, {{ session["username"] }}! | <a href="{{ url_for("logout") }}">Logout</a></div>
+                {% else %}
+                    <meta http-equiv="refresh" content="0; url={{ url_for('login') }}">
```

Comment:
nit: all business logic should be inside views, templates mainly are used for rendering data